src/AasxServerDB/EntityFW.cs

```diff
@@ -0,0 +1,323 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Extensions;
+using System.ComponentModel.DataAnnotations.Schema;
+
+/*
+ * https://learn.microsoft.com/en-us/ef/core/get-started/overview/first-app?tabs=netcore-cli
+ * 
+ * Initial Migration
+ * Add-Migration InitialCreate -Context SqliteAasContext -OutputDir Migrations\Sqlite
+ * Add-Migration InitialCreate -Context PostgreAasContext -OutputDir Migrations\Postgres
+ * 
+ * Change database
+ * Add-Migration XXX -Context SqliteAasContext
+ * Add-Migration XXX -Context PostgreAasContext
+ * Update-Database -Context SqliteAasContext
+ * Update-Database -Context PostgreAasContext
+ */
+
+namespace AasxServerDB
+{
+    public class AasContext : DbContext
+    {
+        public static IConfiguration _con { get; set; }
+        public static string _dataPath { get; set; }
+        public static bool _isPostgres { get; set; }
+
+        public DbSet<AASXSet> AASXSets { get; set; }
+        public DbSet<AASSet> AASSets { get; set; }
+        public DbSet<SMSet> SMSets { get; set; }
+        public DbSet<SMESet> SMESets { get; set; }
+        public DbSet<SValueSet> SValueSets { get; set; }
+        public DbSet<IValueSet> IValueSets { get; set; }
+        public DbSet<DValueSet> DValueSets { get; set; }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder options)
+        {
+            if (_con == null)
+            {
+                throw new Exception("No Configuration!");
+            }
+            string connectionString = _con["DatabaseConnection:ConnectionString"];
+            if (connectionString != null)
+            {
+                if (connectionString.Contains("$DATAPATH"))
+                    connectionString = connectionString.Replace("$DATAPATH", _dataPath);
+                if (connectionString.ToLower().Contains("host")) // Postgres
+                {
+                    string[] Params = connectionString.Split(";");
+                    string dbPassword = Environment.GetEnvironmentVariable("DATABASE_PASSWORD");
+                    string dbUser = Environment.GetEnvironmentVariable("DATABASE_PASSWORD");
+                    for (int i = 0; i < Params.Length; i++)

```

Comment:
duplicate logic for changing parameters should be extracted into another method:

```chsarp
string ReplaceConnectionStringParam(string connectionString, string paramName, string paramValue)
{
    string[] parts = connectionString.Split(';');
    for (int i = 0; i < parts.Length; i++)
    {
        if (parts[i].StartsWith(paramName + "=", StringComparison.OrdinalIgnoreCase))
        {
            parts[i] = paramName + "=" + paramValue;
            break;
        }
    }
    return string.Join(";", parts);
}
```